vectors/no_resizable_vec.rs

```diff
@@ -1,108 +1,93 @@
-use std::cmp::Ordering;
-use cvt::CVT_assume;
+use std::{ptr, ops::{Deref, DerefMut, Index, IndexMut}};
+use alloc::alloc::{Layout, alloc, dealloc};
 use nondet::Nondet;
 use std::io::Read;
 use anchor_lang::prelude::borsh::maybestd::io::Write;
 use borsh::{BorshDeserialize, BorshSerialize};
 
-////////////////////////////////////////////////////////////////////////
-// Adapted from SeaHorn.
-//
-// This is a more precise implementation of the Vector class than
-// NoDataVec but the vector is not resizable so its capacity must be
-// fixed a priori.
-////////////////////////////////////////////////////////////////////////
-
-trait IsZST { const IS_ZST: bool; }
-impl<T> IsZST for T { const IS_ZST: bool = mem::size_of::<T>() == 0; }
-
-macro_rules! assert {
-        ($cond:expr) => {{ /*cvt::CVT_assert($cond)*/}};
-}
-
+/////////////////////////
+/// Raw Vec
+/////////////////////////
 struct RawVec<T> {
     ptr: NonNull<T>,
     cap: usize,
 }
 
-unsafe impl<T: Send> Send for RawVec<T> {}
-unsafe impl<T: Sync> Sync for RawVec<T> {}
-
 impl<T> RawVec<T> {
-    // We assume that there is no errors during allocation.
-    fn new(capacity: usize) -> Self {
-        if T::IS_ZST || capacity == 0 {
-            return RawVec {
-                ptr: NonNull::dangling(),
-                cap: capacity,
-            };
+    fn new_zero_sized() -> Self {
+        Self {
+            ptr: NonNull::dangling(),
+            cap: 0,
         }
+    }
+
+    fn new(capacity: usize) -> Self {
 
-        let layout: Layout = Layout::array::<T>(capacity).unwrap();
-	    //CVT_assume(layout.size() <= isize::MAX as usize);
+        // ZSTs have no memory allocation
+        if mem::size_of::<T>() == 0 || capacity == 0 {
+            return RawVec::new_zero_sized();
+        }
 
-        let pointer: *mut u8 = unsafe { alloc(layout) };
-        let pointer: NonNull<T> = match NonNull::new(pointer as *mut T) {
-            Some(p) => p,
-            None => {
-                CVT_assume(false);
-                handle_alloc_error(layout)
-            },
+        let layout: Layout = Layout::array::<T>(capacity).unwrap_or_else(|_| panic!("capacity overflow"));
+        let ptr: NonNull<T> = unsafe {
+            let ptr: *mut u8 = alloc(layout);
+            NonNull::new_unchecked(ptr as *mut T)
         };
 
-        RawVec {
-            ptr: pointer,
+        Self {
+            ptr,
             cap: capacity,
         }
     }
 }
 
 impl<T> Drop for RawVec<T> {
     fn drop(&mut self) {
-        if self.cap != 0 && !T::IS_ZST {
-            unsafe {
-                dealloc(
-                    self.ptr.as_ptr() as *mut u8,
-                    Layout::array::<T>(self.cap).unwrap(),
-                );
-            }
+        // ZSTs have no memory allocation
+        if mem::size_of::<T>() == 0 {
+            return;
+        }
+
+        let layout: Layout = Layout::array::<T>(self.cap).unwrap_or_else(|_| panic!("capacity overflow"));
```

Comment:
how can `capacity` overflow during `drop`?

Reply:
ya it shouldn't cause a vector with such a capacity wouldn't be alloced in first place.

Removed the error checking